zerver/lib/upload.py

```diff
@@ -844,6 +867,9 @@ def upload_avatar_image(user_file: File, acting_user_profile: UserProfile,
     upload_backend.upload_avatar_image(user_file, acting_user_profile,
                                        target_user_profile, content_type=content_type)
 
+def upload_avatar_image_from_url(url: str, user_profile: UserProfile) -> str:
```

Comment:
I think we only need this function, and can rid of the class methods? The logic of downloading the image is independent of the upload backend. So this function can get the image and then call `upload_avatar_image`, which takes care of using the correct upload backend.